osdep/win32/include/pthread.h

```diff
@@ -16,11 +16,17 @@
 #ifndef MP_WRAP_PTHREAD_H_
 #define MP_WRAP_PTHREAD_H_
 
+// this is MinGW specific and sort of a hack but we really need to prevent
+// system pthread headers from being included, which can happen through <time.h>
+// and <unistd.h>:
+#define WIN_PTHREADS_TIME_H
+#define WIN_PTHREADS_UNISTD_H
```

Comment:
inb4 someone has different pthread implementation in sysroot. Anyway, I will not discuss which hacks are more prone to breaking.